### PR TITLE
Develop jump code for Roman (RCAL-81)

### PR DESCRIPTION
**Pull request by dmggh**
Closes #

Resolves RCAL-81

**Description**
This PR addresses RCAL-81: Develop jump detection code for Roman WFI data

Checklist
- [ x] Tests

- [ x] Documentation

- [ ] Change log

- [ x] Milestone

- [x ] Label(s)

### CODE REVIEW

**Review by ddavis-stsci — COMMENTED** [review]
Which of these routines should go onto STCAL?

File: romancal/jump/jump_step.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    """
+
+    reference_file_types = ['gain', 'readnoise']
+
+    def process(self, input):
+
+       # Open input as a Roman DataModel (single integration; 3D arrays)
+        with datamodels.RampModel(input) as R_input_model:
+            # Create a JWST DataModel (4D arrays) to populate
+            input_model = jwst.datamodels.RampModel()
```
[the remark this answers is not in the document]
Comment (dmggh) [reply romancal/jump/jump_step.py:42]: The jump step code operates on 4D arrays (like ramp fitting)

File: romancal/jump/jump_step.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                                              R_input_model.err.shape)
+
+            input_model.refout = np.broadcast_to(R_input_model.refout, (1,) +
+                                                 R_input_model.refout.shape)
+
+            tstart = time.time()
+            # Check for an input model with NGROUPS<=2
+            ngroups = input_model.data.shape[1]
+            if ngroups <= 4:
+                self.log.warning('Can not apply jump detection when NGROUPS<=4;')
```
[the remark this answers is not in the document]
Comment (dmggh) [reply romancal/jump/jump_step.py:64]: I will fix that comment.

**Comment by nden** [comment]
This code does not use stcal. Is it expected to be reworked?

**Comment by dmggh** [comment]
Yes, reworking it.


________________________________

**Comment by nden** [comment]
#244 was merged. Can this be closed?

**Comment by dmggh** [comment]
Yes, this can be closed.

**Comment by nden** [comment]
@dmggh Please close this if not current.

**Comment by dmggh** [comment]
Superseded by PR#244 so closing

**Pull request closed without merging**